src/main/scala/org/apache/spark/sql/delta/DeltaErrors.scala

```diff
@@ -613,6 +613,15 @@ abstract class DeltaConcurrentModificationException(message: String)
   def conflictType: String = this.getClass.getSimpleName.stripSuffix("Exception")
 }
 
+/**
+ * Thrown when a concurrent transaction has written data to conflicting partitions
+ * after the current transaction read the table.
+ */
+class ConcurrentPartitionWriteException(
```

Comment:
I personally don't think we need this; we can just use the existing exceptions depending on what kind of conflict it was.

Reply:
My intension is to inform the user which partitions were in conflict - `val conflictingPartitions: Iterable[Map[String, String]]` and I couldn't find existing exception class where I could pass this info. 
My personal use case - I'm running a stream which in 99% modifies the last 3 partitions (date col). Then I'm running a compactor which compacts everything but the last 3 partitions. So in 99% they are running without concurrency error since they are operating on independent partitions. But sometimes compactor failed and in this situation I was not sure what happened. Adding `conflictingPartitions` to exception gave me more insight which late arriving partitions caused the conflict.